### PR TITLE
Fixes #1220 Do not let catch escape

### LINKED ISSUE #1220: No response if use 'Select from gallery' after access to photos was denied
### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: There is no response from the app when user taps on Select from gallery to choose profile picture after access to photos was denied. Expected: same as for camera, there is popup wth explanation where to turn on access to photos. 

#### Expected behavior
[comment]: # (Describe what you expected to happen.)
After user denied access to photos on 1st attempt to select pictures from Gallery and repeats this attempt, app should show popup with explanation how to turn on access to photos
#### Actual behavior
[comment]: # (Describe what actually happened.)
Nothing happens when user taps on Select from Gallery
### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)
Video: https://drive.google.com/open?id=0Bz3t9zSg1wb7c2x1WHRGTTloZUE
- Open Status first time
- Open Profile
- Edit Profile
- Edit profile picture
- tap on Select from Gallery. As a result, popup is shown "Deny/Allow" access to photos
- Deny access to photos
- Tap on Select from Gallery again. 

### Additional Information
[comment]: # (Please do your best to fill this out.)
* Status version: 0.9.7
* Operating System: iOS only
OS: iOS 10.3.1
Device: iPhone 7 plus

### PR DESCRIPTION
**Pull request by jeluard**
fix #1220 No response if use 'Select from gallery' after access to photos was denied
 
### Steps to test:
- Open Status first time
- Open Profile
- Edit Profile
- Edit profile picture
- tap on Select from Gallery. As a result, popup is shown "Deny/Allow" access to photos
- Deny access to photos
- Tap on Select from Gallery again.
- An error message is  shown

status: ready

### CODE REVIEW

**Comment by flexsurfer** [comment]
hi @jeluard could you please change the description, looks like summary and steps to test from different tasks. and there is no link on the issue in the description , thank you

**Comment by jeluard** [comment]
My bad the description was wrong.

File: src/status_im/components/permissions.cljs [hunk 1]
```diff
@@ -19,10 +19,12 @@
...
 (defn request-permissions [permissions then else]
   (if platform/android?
-    (let [permissions (mapv #(get permissions-map %) permissions)
-          result      (.requestMultiple permissions-class (clj->js permissions))
-          result      (.then result #(if (all-granted? (js->clj %))
-                                       (then)
-                                       (when else (else))))
-          result      (.catch result #(else))])
+    (letfn [(else-fn [] (when else (else)))]
+      (let [permissions (mapv #(get permissions-map %) permissions)
```
Comment (flexsurfer) [inline src/status_im/components/permissions.cljs]: the form is broken, can't be compiled, Indentation should be adjusted

Comment (jeluard) [inline src/status_im/components/permissions.cljs]: My bad I just fixed the indentation.

**Comment by jeluard** [comment]
Fixed `request-permissions`.

**Comment by flexsurfer** [comment]
thanks, @jeluard could you please squash commits

**Comment by jeluard** [comment]
Done

**Pull request merged**